pkg/hubble/parser/threefour/parser.go

```diff
@@ -201,6 +206,27 @@ func (p *Parser) resolveNames(epID uint32, ip net.IP) (names []string) {
 	return nil
 }
 
+func (p *Parser) resolvePolicies(fl *pb.Flow) (policies []string) {
+	if fl.Verdict != pb.Verdict_FORWARDED || fl.PolicyMatchType == monitorAPI.PolicyMatchNone {
+		return
+	}
+
+	npStore := p.storeGetter.GetK8sStore("networkpolicy")
+	nps := npStore.List()
+	policies = make([]string, 0)
+	for _, np := range nps {
```

Comment:
I'm concerned about the scale implications here.

Reply:
I'll try to do benchmarks and reduce amount of matching.